src/util/bip32.rs

```diff
@@ -161,6 +161,28 @@ impl fmt::Display for ChildNumber {
     }
 }
 
+impl FromStr for ChildNumber {
+    type Err = Error;
+
+    fn from_str(inp: &str) -> Result<ChildNumber, Error> {
+        let hardened = inp.chars().last().map_or(false, |l| l == '\'' || l == 'h');
+        let idx = match hardened {
+            true => inp[0..inp.len() - 1].parse().map_err(|_| Error::InvalidChildNumberFormat)?,
+            false => inp.parse().map_err(|_| Error::InvalidChildNumberFormat)?,
+        };
+        let cn = match hardened {
+            true => ChildNumber::Hardened { index: idx },
+            false => ChildNumber::Normal { index: idx },
+        };
+
+        if idx & (1 << 31) == 0 {
```

Comment:
Yeah I tried to avoid changing existing logic. But that might make more sense. I also don't fully like the `InvalidChildNumber` error variant containing a child number. That forces you to create an invalid one for the purpose of creating the error..